content/03_CreatingObservableSequences.md

```diff
@@ -0,0 +1,1274 @@
+
+So you can think of `BehaviorSubject<T>` as a subject that _always_ has a value available. If you subscribe to a `BehaviorSubject<T>` it will instantly produce a single value. (It may then go on to produce more values, but it always produces one right away.) As it happens, it also makes that value available through a property called `Value`, so you don't need to subscribe an `IObserver<T>` to it just to retrieve the value.
+
+A `BehaviorSubject<T>` could be thought of an as observable property. Like a normal property, it can immediately supply a value whenever you ask it. The difference is that it can then go on to notify you every time its value changes.
+
+This analogy falls down slightly when it comes to completion. If you call `OnCompleted`, it immediately calls `OnCompleted` on all of its observers, and if any new observers subscribe, they will also immediately be completed—it does not first supply the last value. (So this is another way in which it is different from a `ReplaySubject<T>` with a buffer size of 1.)
+
+Similarly, if you call `OnError`, all current observers will receive an `OnError` call, and any subsequent subscribers will also receive nothing but an `OnError` call.
+
+## AsyncSubject<T>
+
+`AsyncSubject<T>` provides all observers with the final value it receives. Since it can't know which is the final value until `OnCompleted` is called, it will not invoke any methods on any of its subscribers until either its `OnCompleted` or `OnError` method is called. (If `OnError` is called, it just forwards that to all current and future subscribers.)
+
+If no calls were made to `OnNext` before `OnCompleted` then there was no final value, so it will just complete any observers without providing a value.
+
+In this example no values will be published as the sequence never completes. 
+No values will be written to the console.
+
+```csharp
+AsyncSubject<string> subject = new();
+subject.OnNext("a");
+subject.Subscribe(x => Console.WriteLine($"Sub1: {x}"));
+subject.OnNext("b");
+subject.OnNext("c");
+```
+
+In this example we invoke the `OnCompleted` method so there will be a final value ('c') for the subject to produce:
+
+```csharp
+AsyncSubject<string> subject = new();
+
+subject.OnNext("a");
+subject.Subscribe(x => Console.WriteLine($"Sub1: {x}"));
+subject.OnNext("b");
+subject.OnNext("c");
+subject.OnCompleted();
+subject.Subscribe(x => Console.WriteLine($"Sub2: {x}"));
+```
+
+This produces the following output:
+
+```
+Sub1: c
+Sub2: c
+```
+
+## Subject factory
```

Comment:
A real-world scenario would be useful

Reply:
Again, I can't tell which of the two sections spanned by this highlight you mean. If it's `AsyncSubject<T>` (which is what the first 3 lines are from) I've added one.

If it's about the subject factory, this is another case where I don't really know what this is for. It's a section I've retained from Lee's book. I've added some detail because it wasn't even clear to me what it did from his original text.

I'm slightly reluctant to remove it simply because for many years I've had a suspicion that there's some epiphany around subjects that has thus far eluded me. I know mechanically what they do and I know some uses for some of the concrete ones. But I never really understood why they are apparently imbued with particular significance in Reaqtor.